pandas/core/arrays/arrow/array.py

```diff
@@ -1650,6 +1651,82 @@ def _replace_with_mask(
         result[mask] = replacements
         return pa.array(result, type=values.type, from_pandas=True)
 
+    # ------------------------------------------------------------------
+    # GroupBy Methods
+
+    def _to_masked(self):
+        pa_dtype = self._pa_array.type
+        na_value = 1
+        from pandas.core.arrays import (
+            BooleanArray,
+            FloatingArray,
+            IntegerArray,
+        )
+
+        arr_cls: type[FloatingArray | IntegerArray | BooleanArray]
+        if pa.types.is_floating(pa_dtype):
+            nbits = pa_dtype.bit_width
+            dtype = f"Float{nbits}"
+            np_dtype = dtype.lower()
+            arr_cls = FloatingArray
+        elif pa.types.is_unsigned_integer(pa_dtype):
+            nbits = pa_dtype.bit_width
+            dtype = f"UInt{nbits}"
+            np_dtype = dtype.lower()
+            arr_cls = IntegerArray
+
+        elif pa.types.is_signed_integer(pa_dtype):
+            nbits = pa_dtype.bit_width
+            dtype = f"Int{nbits}"
+            np_dtype = dtype.lower()
+            arr_cls = IntegerArray
+
+        elif pa.types.is_boolean(pa_dtype):
+            dtype = "boolean"
+            np_dtype = "bool"
+            na_value = True
+            arr_cls = BooleanArray
+        else:
+            raise NotImplementedError
+
+        mask = self.isna()
+        arr = self.to_numpy(dtype=np_dtype, na_value=na_value)
+        return arr_cls(arr, mask)
```

Comment:
See my comment on the issue, I think this can be optimized and simplified by reusing `__from_arrow__` (which uses `pyarrow_array_to_numpy_and_mask` under the hood)